### ISSUE TITLE
Entity:GetSequenceInfo() is broken on viewmodels

### ISSUE THREAD

**Issue by TiberiumFusion** [Facepunch/garrysmod-issues#4431]
### Details
`GetSequenceInfo()` returns incorrect data. Specifically, [`SequenceInfo`](https://wiki.facepunch.com/gmod/Structures/SequenceInfo)`.activity` is always -1 for every single sequence that the model has. The other fields are correct. Obviously, this is a huge problem when it comes to making SWEPs.

It gets weirder. If the viewmodel is a swep and it performs any kind of action, (like a `DefaultReload()`, a `PrimaryAttack()`, switching to another weapon then switching back, etc.) then `GetSequenceInfo()` starts returning correct data. In other words, [`SequenceInfo`](https://wiki.facepunch.com/gmod/Structures/SequenceInfo)`.activity` will be correct ONLY after the player makes the swep do something with its viewmodel.

In other, other words, picking up a weapon does not fully "initialize" (or crawl, or survey, etc) the viewmodel.

And it gets even weirder. It's only clientside. `GetSequenceInfo()` consistently returns correct data on the server as one would expect.

Weapons that you spawn with are not affected. HL2 weapons are not affected. This only seems to affect workshop weapons that are picked up, equipped from the spawnmenu, etc.

This issue literally appeared out of thin air at some point in the last 2-3 days. `GetSequenceInfo()` worked perfectly fine on the 26th, but as of today it is now broken. All of the addons I'm working with have not been updated for several months. Nothing changed in my setup, aside from joining the exact same servers with the exact same addons I've been joining from before the bug appeared.

### Steps to reproduce
1. Get yourself some weapons from any addon. I've tested with the M9K guns, the included DarkRP guns, and my own SWEPs, including super stripped-down ones. It reproduces each time on all of them.
2. Put this code somewhere:
```
local function check()
	print("\n\n\n========================")
	local lp = LocalPlayer()
	if (IsValid(lp)) then
		local vm = lp:GetViewModel()
		if (vm ~= nil) then
			for seqId, seqName in pairs(vm:GetSequenceList()) do
				local seqInfo = vm:GetSequenceInfo(seqId)
				print("===== Sequence:   ", seqId, seqName)
				PrintTable(seqInfo)
			end
		end
	end
	print("\n\n")
end

concommand.Add("ActivityIDTest", check)
```

3. Equip a weapon from the spawnmenu
4. Run ActivityIDTest in your console
5. Notice that all the `.activity` fields are -1
6. Shoot the gun, or reload it, or switch away then back again, or do whatever to make the viewmodel play a sequence.
7. Run ActivityIDTest in your console again
8. Notice that all the `.activity` fields are finally correct

### Environment
I've confirmed this on vanilla 2020.03.21 and on x86-64 2020.03.25. But again, this issue appeared out of thin air 2-3 days ago when everything was perfectly fine just before **on the exact same versions with the exact same addons** (ofc I could be wrong but I'm 99% sure on this). This makes me suspect a behind-the-scenes workshop change or a complex, adhoc gmod bug.

And yes, I validated my game files several times and reinstalled the addons I was testing with. The bug still persists.

**Comment by robotboy655** [comment]
> This issue literally appeared out of thin air at some point in the last 2-3 days. GetSequenceInfo() worked perfectly fine on the 26th, but as of today it is now broken.

Yeah, this isn't how that works. There wasn't any changes to the live game since the hotfix. It either was there always or it wasn't.

It seems the activity field is only populated if you actually play an animation on the model first, specifically with `SelectWeightedSequence` or whatever calls into it, such as `SentWeaponAnim`

**Comment by robotboy655** [comment]
The underlying issue as to why this happens in the note here:
https://wiki.facepunch.com/gmod/WEAPON:Deploy

I will update GetSequenceInfo to fill the activity field when it is called so this should no longer be an issue.

**Comment by TiberiumFusion** [comment]
@robotboy655 
>The underlying issue as to why this happens in the note here:
https://wiki.facepunch.com/gmod/WEAPON:Deploy

I'm assuming you are referencing "in multiplayer it will not be called clientside if the weapon is switched with Player:SelectWeapon or the "use" console command" right?

Maybe the issue is deeper than that. Even if Deploy() *is* called on the client, like so...
```
function SWEP:Initialize()
	if CLIENT then
		self:Equip(self.Owner)
		self:Deploy()
	end
end
```
...then `GetSequenceInfo().activity` is still equal to `-1` until a viewmodel sequence plays.


>It either was there always or it wasn't.

Yes, this is why I postulated "a behind-the-scenes workshop change or a complex, adhoc gmod bug." Complex & adhoc as in, a bug introduced some time ago that takes several iterations of some process to fully manifest. For example, a config value that increments with each game launch, and when that value exceeds some threshold, buggy behavior starts to occur.

I only mentioned this factor because I took complete, nightly images of the code I was working on. The bug reproduced on the same image that was fine a few days ago. If that behavior is relevant to the bug symptoms, then it could be extremely valuable information in getting a lead on locating a complex bug with delayed manifestation. So, at the risk of someone thinking I'm an idiot, I chose to share that information. I believe that a complete report is always better than an incomplete one.

**tl;dr** By "complex, adhoc gmod bug" I was indeed referring to a possible bug that "was there always"

**Comment by robotboy655** [comment]
That code won't achieve anything unless you play an animation (Specifically with ACT_ enums) from `SWEP:Deploy` which isn't the case by default.

Calling it like that won't run the C++ side of things, which plays the default deploy animation, you are only calling the Lua function itself.
